openstudiocore/src/utilities/filetypes/EpwFile.cpp

```diff
@@ -132,67 +132,119 @@ EpwDataPoint::EpwDataPoint(int year,int month,int day,int hour,int minute,
-  pt.setLiquidPrecipitationQuantity(list[EpwDataField::LiquidPrecipitationQuantity].toStdString());
+  if (!pt.setYear(list[EpwDataField::Year])) {
+    return boost::none;
+  }
+  if (!pt.setMonth(list[EpwDataField::Month])) {
+    return boost::none;
+  }
+  if (!pt.setDay(list[EpwDataField::Day])) {
+    return boost::none;
+  }
+  if (!pt.setHour(list[EpwDataField::Hour])) {
+    return boost::none;
+  }
+  if(!pt.setMinute(list[EpwDataField::Minute])) {
+    return boost::none;
+  }
+  pt.setDataSourceandUncertaintyFlags(list[EpwDataField::DataSourceandUncertaintyFlags]);
+  pt.setDryBulbTemperature(list[EpwDataField::DryBulbTemperature]);
+  pt.setDewPointTemperature(list[EpwDataField::DewPointTemperature]);
+  pt.setRelativeHumidity(list[EpwDataField::RelativeHumidity]);
+  pt.setAtmosphericStationPressure(list[EpwDataField::AtmosphericStationPressure]);
+  pt.setExtraterrestrialHorizontalRadiation(list[EpwDataField::ExtraterrestrialHorizontalRadiation]);
+  pt.setExtraterrestrialDirectNormalRadiation(list[EpwDataField::ExtraterrestrialDirectNormalRadiation]);
+  pt.setHorizontalInfraredRadiationIntensity(list[EpwDataField::HorizontalInfraredRadiationIntensity]);
+  pt.setGlobalHorizontalRadiation(list[EpwDataField::GlobalHorizontalRadiation]);
+  pt.setDirectNormalRadiation(list[EpwDataField::DirectNormalRadiation]);
+  pt.setDiffuseHorizontalRadiation(list[EpwDataField::DiffuseHorizontalRadiation]);
+  pt.setGlobalHorizontalIlluminance(list[EpwDataField::GlobalHorizontalIlluminance]);
+  pt.setDirectNormalIlluminance(list[EpwDataField::DirectNormalIlluminance]);
+  pt.setDiffuseHorizontalIlluminance(list[EpwDataField::DiffuseHorizontalIlluminance]);
+  pt.setZenithLuminance(list[EpwDataField::ZenithLuminance]);
+  pt.setWindDirection(list[EpwDataField::WindDirection]);
+  pt.setWindSpeed(list[EpwDataField::WindSpeed]);
+  pt.setTotalSkyCover(list[EpwDataField::TotalSkyCover]);
+  pt.setOpaqueSkyCover(list[EpwDataField::OpaqueSkyCover]);
+  pt.setVisibility(list[EpwDataField::Visibility]);
+  pt.setCeilingHeight(list[EpwDataField::CeilingHeight]);
+  pt.setPresentWeatherObservation(list[EpwDataField::PresentWeatherObservation]);
+  pt.setPresentWeatherCodes(list[EpwDataField::PresentWeatherCodes]);
+  pt.setPrecipitableWater(list[EpwDataField::PrecipitableWater]);
+  pt.setAerosolOpticalDepth(list[EpwDataField::AerosolOpticalDepth]);
+  pt.setSnowDepth(list[EpwDataField::SnowDepth]);
+  pt.setDaysSinceLastSnowfall(list[EpwDataField::DaysSinceLastSnowfall]);
+  pt.setAlbedo(list[EpwDataField::Albedo]);
+  pt.setLiquidPrecipitationDepth(list[EpwDataField::LiquidPrecipitationDepth]);
+  pt.setLiquidPrecipitationQuantity(list[EpwDataField::LiquidPrecipitationQuantity]);
   return boost::optional<EpwDataPoint>(pt);
 }
 
-boost::optional<std::string> EpwDataPoint::unitsByName(std::string name)
+std::vector<std::string> EpwDataPoint::toEpwStrings() const
+{
+  std::vector<std::string> list;

```

Comment:
Vector is fine, just reserve 35 elements when you create it so it doesn't have to do a resize later.  Or even better, use the size constructor and then the enums to populate it:

    std::vector<std::string> list(35,"");
    list[EpwDataField::Year] = std::to_string(m_year);